app/scripts/lib/transaction/institutional-snap/deferred-publication-hooks.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  TransactionController,
+  TransactionEnvelopeType,
+  TransactionMeta,
+  TransactionStatus,
+} from '@metamask/transaction-controller';
+import type { HandleSnapRequest } from '@metamask/snaps-controllers';
+import { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+import { BaseController, RestrictedMessenger } from '@metamask/base-controller';
+import InstitutionalWalletSnap from '@metamask/institutional-wallet-snap/dist/preinstalled-snap.json';
+import { AccountsControllerGetAccountByAddressAction } from '@metamask/accounts-controller';
+
+const snapId = InstitutionalWalletSnap.snapId as SnapId;
+
+type SnapRPCRequest = Parameters<HandleSnapRequest['handler']>[0];
+
+// Todo: obtain this type from the snap package
+export type InstitutionalSnapResponse = {
+  keyringRequest: {
+    id: string;
+    scope: string;
+    account: string;
+    request: {
+      method: string;
+      params: [
+        {
+          chainId: string;
+          nonce: string;
+          maxPriorityFeePerGas: string;
+          maxFeePerGas: string;
+          gasLimit: string;
+          to: string;
+          value: string;
+          data: string;
+          accessList: string[];
+          from: string;
+          type: string;
+        },
+      ];
+    };
+  };
+  type: string;
+  fulfilled: boolean;
+  rejected: boolean;
+  lastUpdated: number;
+  transaction: {
+    custodianTransactionId: string;
+    transactionStatus: {
+      finished: boolean;
+      success: boolean;
+      displayText: string;
+      submitted: boolean;
+      reason: string;
+      signed: boolean;
+    };
+    from: string;
+    custodianPublishesTransaction: boolean;
+    maxFeePerGas: string;
+    maxPriorityFeePerGas: string;
+    gasLimit: string;
+    nonce: string;
+    to: string;
+    transactionHash: string;
+    type: string;
+  };
+  result: {
+    v: string;
+    r: string;
+    s: string;
+  };
+};
+
+export type InstitutionalSnapRequestSearchParameters = {
+  from: string;
+  to: string;
+  value: string;
+  data: string;
+  chainId: string;
+};
+
+type AllowedActions =
+  | HandleSnapRequest
+  | AccountsControllerGetAccountByAddressAction;
+
+export type DeferredPublicationHookMessenger = RestrictedMessenger<
```

Comment:
As part of the modular controller init pattern, we define the controller messengers in `app/scripts/controller-init/messengers` so the platform team can oversee any new permissions  added to a controller via the messenger.